src/decode/lzma.rs

```diff
@@ -559,3 +580,54 @@ impl DecoderState {
         Ok(result)
     }
 }
+
+#[derive(Debug)]
+/// Raw decoder for LZMA.
+pub struct LzmaDecoder {
+    params: LzmaParams,
+    memlimit: usize,
+    decoder: DecoderState,
```

Comment:
Nit: a "decoder" within a decoder sounds weird, it might be worth calling this variable "state" (or "decoder_state") instead?